internal/testrunner/runners/system/runner.go

```diff
@@ -554,6 +559,10 @@ func filterAgents(allAgents []kibana.Agent, ctx servicedeployer.ServiceContext)
 
 	var filtered []kibana.Agent
 	for _, agent := range allAgents {
+		if agent.PolicyRevision == 0 {
+			continue // For some reason Kibana doesn't always return a valid policy revision (eventually it will be present and valid)
```

Comment:
Not blocking this PR but it would be good to understand why this is happening and try to remove this condition in a follow up PR. @nchaulet maybe you can shed some light here?

Reply:
This is actually causing some flakiness for Kubernetes integration. I would be for leaving it here (to unblock our delivery pipeline), but it definitely needs an explanation.

Reply:
Is it by design or it just works this way? I'm concerned if it makes sense to have an agent with policy assigned, but no revision. Shouldn't it always be 1 on startup?